.github/workflows/npm-publish.yml

```diff
@@ -9,23 +9,21 @@ on:
     inputs:
       fetch-depth:
         description: Number of commits to fetch. 0 indicates all history for all branches and tags.
-        default: 1
-        type: number
+        default: "1"
+        type: string
```

Comment:
[fetch-depth](https://github.com/actions/checkout/blob/163217dfcd28294438ea1c1c149cfaf66eec283e/action.yml#L74) defaults to 1 (so it's defined as a number). I think we should keep it that way

Reply:
In the actions, the inputs are stored as environment variables (strings):

https://docs.github.com/en/actions/sharing-automations/creating-actions/metadata-syntax-for-github-actions#inputs

And in fact the action code converts is to a `Number` just after getting it from the inputs/environment:

https://github.com/actions/checkout/blob/v4.2.1/src/input-helper.ts#L105

If this is a breaking change to any user of the workflow, IMO we should fix that code instead.

Reply:
🤝